src/index.ts

```diff
@@ -195,12 +203,11 @@ export default class Requester {
     return false;
   };
 
-  /**
-   * This method replace the values of the given object
-   */
-  private changeAuthorizationHeader = (options: AxiosRequestConfig): void => {
-    options["headers"] = {
-      ...options["headers"],
+  private replaceAxiosRequestAuthorizationHeader = (
+    headers: object
+  ): object => {
+    return {
+      ...headers,
```

Comment:
I think that it is better for you to use ({ headers }) at the function parameter instead of passing options['headers'] as parameter

Reply:
Nice! Changed it.

Reply:
Also, I've found another bug when the request was trying to authenticate inside the CATCH method.